### PR TITLE
[dockerng] WIP No more runtime args passed to docker.start()

### LINKED ISSUE #27661: The dockerng module uses deprecated API in docker-py
According to the docker-py 1.4 API documentation, providing arguments to `client.start` is decrecated. These arguments should be provided to the create method instead. This follows from a similiar change in the Docker API 1.15. We're now at 1.20 so it's high time to switch. 

A consequence of using the deprecated API is that you can't provide LogConfig arguments to docker. The reason is that the only way to provide this with docker-py is in the `host_config` provided to `client.create` however, the `host_config` given to `client.start` overrides these. 

```
Deprecation warning: For API version > 1.15, it is highly recommended to provide host config options in the host_config parameter of create_container
```

### PR DESCRIPTION
**Pull request by ticosax**
fixes #27661 
I open this pull request for discussion.
Some tests for `modules.dockerng` are missing.

/cc @terminalmage @alf

### CODE REVIEW

File: salt/modules/dockerng.py [hunk 1]
```diff
@@ -4168,174 +4304,13 @@ def signal_(name, signal):
 
 @_refresh_mine_cache
 @_ensure_exists
-def start(name, validate_ip_addrs=True, **kwargs):
+def start(name, **kwargs):
```
Comment (terminalmage) [inline salt/modules/dockerng.py]: We can drop the ``**kwargs`` here, since we are no longer doing anything with them.

**Comment by cachedout** [comment]
@ticosax This has merge conflicts. Could you please rebase it?

**Comment by ticosax** [comment]
Thanks @cachedout 
@terminalmage I addressed your comment.
 When I wrote a new test I realized it is already covered. Since every extra_argument that moves from `start()` to `create()` is passed in host_config dictionary. Which support has been added by #27451.

So, if tests are passing I believe it is good to go.

**Comment by terminalmage** [comment]
Great, I'll review it today.

**Comment by ticosax** [comment]
a fix for the test is coming

**Comment by ticosax** [comment]
@terminalmage I recommend to review a first pass with https://github.com/saltstack/salt/pull/27676/files?w=1
As big part of changes are just re-indentation.

**Comment by terminalmage** [comment]
This looks great and the one failing test on the Cent7 side is unrelated.

**Comment by ticosax** [comment]
@terminalmage thank you.

**Pull request merged**